lib/player.js

```diff
@@ -5586,6 +5621,38 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     this.dispatchEvent(this.makeEvent_(eventName, {detail: clone}));
   }
 
+  /**
+   * When notified of a media quality change we need to emit a
+   * MediaQualityChange event to the app.
+   *
+   * @param {shaka.extern.MediaQualityInfo} mediaQuality
+   * @param {number} position
+   *
+   * @private
+   */
+  onMediaQualityChange_(mediaQuality, position) {
+    // Always make a copy to avoid exposing our internal data to the app.
+    const clone = {
+      bandwidth: mediaQuality.bandwidth,
+      audioSamplingRate: mediaQuality.audioSamplingRate,
+      codecs: mediaQuality.codecs,
+      contentType: mediaQuality.contentType,
+      frameRate: mediaQuality.frameRate,
+      height: mediaQuality.height,
+      mimeType: mediaQuality.mimeType,
+      channelsCount: mediaQuality.channelsCount,
+      pixelAspectRatio: mediaQuality.pixelAspectRatio,
+      width: mediaQuality.width,
+    };
+
+    const data = {
+      'mediaQuality': clone,
```

Comment:
I found that the `mediaQuality` property name was being minified to `re` in the build process unless wrapped in quotation marks. Any idea why that happens?

Reply:
Yeah, that's a known issue. If you don't specify the dictionary values as strings, Closure Compiler minifies them. It was reported a few weeks ago as #3710.
Actually, I made a CL pretty recently that enforces conformance on this front by changing the expected type of this data object from `Object` to `Map.<string, Object>`. See https://github.com/google/shaka-player/commit/dd3748d59970de46255cd6d8b4204ee501600655. You're probably going to have to rebase onto that, now that I think about it.

Reply:
Any object that isn't declared as part of an "extern" interface, the compiler will rename its fields.  It's sometimes not worth creating definitions for everything, so quoted property names are the lightweight option.